eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/RelationConverter.php

```diff
@@ -66,30 +67,103 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
      */
     public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef)
     {
-        // Selection method, 0 = browse, 1 = dropdown
-        $storageDef->dataInt1 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionMethod'];
+        $fieldSettings = $fieldDef->fieldTypeConstraints->fieldSettings;
+        $doc = new DOMDocument('1.0', 'utf-8');
+        $root = $doc->createElement('related-objects');
+        $doc->appendChild($root);
 
-        // Selection root, location ID
-        $storageDef->dataInt2 = (int)$fieldDef->fieldTypeConstraints->fieldSettings['selectionRoot'];
+        $constraints = $doc->createElement('constraints');
+        if (!empty($fieldSettings['selectionContentTypes'])) {
+            foreach ($fieldSettings['selectionContentTypes'] as $typeIdentifier) {
```

Comment:
@bdunogier Are you suggesting we should do something like `$storageDef->dataText5 = $this->getXMLForStorageFieldDefinition($fieldDef->fieldTypeConstraints->fieldSettings);`, or something else?

Implied here is that he will need to adopt RelationListConverter as well, part of the point here is to keep them in sync. Could perhaps also extract some of the logic to `RelationConverterTrait` and reuse, wdyt @slaci ?